packages/@react-stately/dnd/src/useDroppableCollectionState.ts

```diff
@@ -29,38 +30,93 @@ export interface DroppableCollectionState {
 }
 
 export function useDroppableCollectionState(props: DroppableCollectionStateOptions): DroppableCollectionState  {
+  let {
+    acceptedDragTypes = 'all',
+    onInsert,
+    onRootDrop,
+    onItemDrop,
+    onReorder,
+    isValidDropTarget,
+    collection,
+    selectionManager,
+    onDropExit,
+    onDropEnter,
+    getDropOperation,
+    onDrop
+  } = props;
   let [target, setTarget] = useState<DropTarget>(null);
 
   let getOppositeTarget = (target: ItemDropTarget): ItemDropTarget => {
     if (target.dropPosition === 'before') {
-      let key = props.collection.getKeyBefore(target.key);
+      let key = collection.getKeyBefore(target.key);
       return key != null ? {type: 'item', key, dropPosition: 'after'} : null;
     } else if (target.dropPosition === 'after') {
-      let key = props.collection.getKeyAfter(target.key);
+      let key = collection.getKeyAfter(target.key);
       return key != null ? {type: 'item', key, dropPosition: 'before'} : null;
     }
   };
 
+  let defaultGetDropOperation = useCallback((target: DropTarget, types: DragTypes, allowedOperations: DropOperation[]) => {
+    // TODO: will be covered in a separate PR (issue in board), for now ignore the typescript error
+    // @ts-ignore
+    let typesSet = types.types ? types.types : types;
+    let draggedTypes = [...typesSet.values()];
+    let {draggingCollectionRef, currentDropCollectionRef, draggingKeys} = getDnDState();
+    // Compare draggingCollectionRef to currentDropCollectionRef since droppedCollectionRef hasn't been set yet (drop hasn't happened)
+    let isInternalDrop = draggingCollectionRef?.current === currentDropCollectionRef?.current;
+
+    if (
+      (acceptedDragTypes === 'all' || draggedTypes.every(type => acceptedDragTypes.includes(type))) &&
```

Comment:
look into making this `.some`, will need to filter stuff out in onDrop.

Reply:
For now IMO we can pass the onus of filtering the actual accepted drop items to the user since we currently expect them to create an item processor function themselves (I imagine they would do the filtering as a part of the item processing). If we wanna help with the filtering in the default onDrop it may look roughly like this:
```
    let filteredItems;
    if (!getDropOperation) {
      filteredItems = items.filter((item) => {
        let itemTypes = [];
        if (item.kind === 'directory') {
          // call getEntries and recusively generate the itemTypes
        } else {
          itemTypes = item.kind === 'file' ? [item.type] : [...item.types];
        }
        
        let validDragType = acceptedDragTypes === 'all' || itemTypes.some(type => acceptedDragTypes.includes(type));
        if (validDragType) {
          return (target.type === 'root' || (target.dropPosition === 'on' && isValidDropTarget(target, item.types)));  
        }
      });
    }
```
where we then provide filteredItems to the util handlers instead of items